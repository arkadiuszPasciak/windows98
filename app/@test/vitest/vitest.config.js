import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
