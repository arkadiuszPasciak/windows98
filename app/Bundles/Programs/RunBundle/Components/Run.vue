<template>
  <UIModal
    class="Run"
    :title="t('RunBundle.title')"
    :resize-window="false"
    width="330px"
    height="165px"
    :modal-state="programStore.modalRun"
    @close-modal="closeModal"
  >
    <div class="content">
      <img
        class="icon"
        width="32"
        height="32"
        src="/icons/applications/run.png"
        alt="Run computer"
        data-test="run-icon"
      >
      <UIText
        class="description"
        data-test="run-description"
      >
        {{ t('RunBundle.description') }}
      </UIText>
      <UIInput
        id="run-input"
        v-model:modelValue="modelValue"
        class="input"
        :label-name="t('RunBundle.open')"
        label-position="left"
      />
      <UIButton
        class="button is-ok"
        data-test="run-button-ok"
        @click="openProgram"
      >
        {{ t('RunBundle.ok') }}
      </UIButton>
      <UIButton
        class="button is-cancel"
        data-test="run-button-cancel"
        @click="closeModal"
      >
        {{ t('RunBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP|Bundles/ProgramBundle/Stores/Program.stores"
import { checkAndRunProgram } from "@APP|Bundles/RunBundle/Services/Run.services"
import UIButton from "@APP|Bundles/UIButtonBundle/Components/UIButton.vue"
import UIInput from "@APP|Bundles/UIInputBundle/Components/UIInput.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UIText from "@APP|Bundles/UITextBundle/Components/UIText.vue"
import { ref } from "vue"
import { useI18n } from "vue-i18n"

const { t } = useI18n()
const programStore = useProgramStore()

const closeModal = (): void => {
	programStore.updateRunModal(false)
}

const openProgram = (): void => {
	checkAndRunProgram(modelValue.value, t)
}

const modelValue = ref<string>("")
</script>

<style lang="scss" scoped src="@APP|Bundles/RunBundle/Styles/Run.styles.scss" />
