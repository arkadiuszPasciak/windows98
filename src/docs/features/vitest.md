# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
