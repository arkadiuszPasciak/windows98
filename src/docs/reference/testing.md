# Testing

This documentation provides an overview of the testing process employed in our application. We utilize two powerful testing tools to ensure the reliability and stability of the code: Cypress and Vitest. These tools are employed in different aspects of testing to maximize coverage and maintain code quality.

## [Vitest](https://vitest.dev/)

`@/vite.config.ts` (in the block test)

Vitest is a lightweight and fast testing framework specifically designed for Vite applications. It provides seamless integration with Vite, supporting both JavaScript and TypeScript code. Vitest is particularly well-suited for unit testing, which focuses on verifying the functionality of individual code units, such as functions, classes, or modules.

We use Vitest for `unit testing` our TypeScript code, ensuring that each function or class in our application performs as intended. This approach allows us to isolate potential issues and fix them before they cause problems in the broader application context.

To effectively use Vitest, we create a series of test suites that group related tests, with each test case focusing on a specific unit of code. These test cases cover various scenarios, such as input validation, error handling, and edge cases, to ensure comprehensive coverage of our codebase.
