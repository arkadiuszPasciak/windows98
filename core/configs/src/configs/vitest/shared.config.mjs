import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
