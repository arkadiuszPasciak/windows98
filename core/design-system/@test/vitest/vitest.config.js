import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
