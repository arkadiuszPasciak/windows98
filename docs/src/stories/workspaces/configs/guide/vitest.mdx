import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

