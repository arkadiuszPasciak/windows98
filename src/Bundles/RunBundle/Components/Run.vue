<template>
  <UIModal
    class="Run"
    :title="t('RunBundle.title')"
    :resize-window="false"
    width="330px"
    height="165px"
    :modal-state="programStore.modalRun"
    @close-modal="closeModal"
  >
    <div class="content">
      <img
        class="icon"
        width="32"
        height="32"
        src="/icons/applications/run.png"
        alt="Run computer"
        data-test="run-icon"
      />
      <UIText class="description" data-test="run-description">
        {{ t('RunBundle.description') }}
      </UIText>
      <UIInput
        id="run-input"
        v-model:modelValue="modelValue"
        class="input"
        :label-name="t('RunBundle.open')"
        label-position="left"
      />
      <UIButton
        class="button is-ok"
        data-test="run-button-ok"
        @click="openProgram"
      >
        {{ t('RunBundle.ok') }}
      </UIButton>
      <UIButton
        class="button is-cancel"
        data-test="run-button-cancel"
        @click="closeModal"
      >
        {{ t('RunBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIInput from '@Bundles/UIBundle/Components/UIInput.vue'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import UIText from '@Bundles/UIBundle/Components/UIText.vue'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import { checkAndRunProgram } from '@Bundles/RunBundle/Services/Run.services'

  const { t } = useI18n()
  const programStore = useProgramStore()

  const closeModal = (): void => {
    programStore.updateRunModal(false)
  }

  const openProgram = (): void => {
    checkAndRunProgram(modelValue.value, t)
  }

  const modelValue = ref('') as Ref<string>
</script>

<i18n src="@Bundles/RunBundle/Locales/Run.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/RunBundle/Styles/Run.styles.scss';
</style>
