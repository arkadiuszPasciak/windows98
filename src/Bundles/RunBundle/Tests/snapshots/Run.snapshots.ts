import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import { createTestingPinia } from '@pinia/testing'
import Run from '@Bundles/RunBundle/Components/Run.vue'

const defaultWrapper = mount(Run, {
  global: {
    plugins: [
      createTestingPinia({
        initialState: {
          program: { modalRun: true },
        },
      }),
    ],
  },
})

describe('[RunBundle]<Snapshots>(Run)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
