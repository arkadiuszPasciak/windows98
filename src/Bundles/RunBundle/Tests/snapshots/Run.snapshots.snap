// Vitest Snapshot v1

exports[`[RunBundle]<Snapshots>(Run) > renders correctly component 1`] = `
<div
  class="UIModal move-window Run"
  data-v-00dd9280=""
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 330px; height: 165px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-test="modal-header-title"
        data-v-99a75817=""
      >
        Run
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    <!--v-if-->
    
    <div
      class="content"
      data-v-00dd9280=""
    >
      <img
        alt="Run computer"
        class="icon"
        data-v-00dd9280=""
        height="32"
        src="/icons/applications/run.png"
        width="32"
      />
      <p
        class="UIText description"
        data-v-00dd9280=""
        data-v-f0b878d5=""
      >
        
        Type of the name of a program and Windows will open it for you
        
      </p>
      <div
        class="UIInput label-position-left input"
        data-v-00dd9280=""
        data-v-e394574e=""
      >
        <label
          class="label"
          data-v-e394574e=""
          for="run-input"
        >
          Open:
        </label>
        <input
          class="input"
          data-v-e394574e=""
          id="run-input"
          type="text"
        />
      </div>
      <button
        class="UIButton size-medium button is-ok"
        data-v-00dd9280=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        OK
        
        
        
      </button>
      <button
        class="UIButton size-medium button is-cancel"
        data-v-00dd9280=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        Cancel
        
        
        
      </button>
    </div>
    
  </div>
</div>
`;
